xrtpy/response/xrt_teem.py

```diff
@@ -0,0 +1,735 @@
+        filt2 = fw1
+        if fw2 != "Open":
+            filt2 = f"{fw1}/{fw2}"
+    elif fw2 != "Open":
+        filt2 = fw2
+    else:
+        raise ValueError("Invalid filter values, both fw1 and fw2 are Open")
+    date_obs2 = hdr2["DATE_OBS"]
+    tresp2 = TemperatureResponseFundamental(filt2, date_obs2)
+
+    if filt1 == filt2:
+        raise ValueError("Filters for the two images cannot be the same")
+
+    T_e, EM, model_ratio, ok_pixel = derive_temperature(
+        map1, map2, tresp1, tresp2, binfac=binfac, Trange=Trange
+    )
+
+    T_error, EMerror, K1, K2 = calculate_TE_errors(
+        map1, map2, T_e, EM, model_ratio, tresp1, tresp2, Trange=Trange
+    )
+
+    T_e = T_e.filled(0.0)
+    EM = EM.filled(0.0)
+    T_error = T_error.filled(0.0)
+    EMerror = EMerror.filled(0.0)
+    if not no_threshold:
+        ok_wothr = ok_pixel.copy()
+        Kd1 = np.sqrt(K1 / data1)
+        Kd1 = Kd1.filled(0.0)
+        Kd2 = np.sqrt(K2 / data2)
+        Kd2 = Kd2.filled(0.0)
+        tthr = (T_error - T_e) <= np.log10(Te_err_threshold)
+        k1thr = Kd1 <= photon_noise_threshold
+        k2thr = Kd2 <= photon_noise_threshold
+        ok_pixel = (
+            ok_pixel
+            & ((T_error - T_e) <= np.log10(Te_err_threshold))
+            & (Kd1 <= photon_noise_threshold)
+            & (Kd2 <= photon_noise_threshold)
+        )
+        if verbose:
+            print(f"number of pixels ruled out by threshold = " f"{np.sum(~ok_pixel)}")
+            print(f"number of pixels ruled out by T_e errors = {np.sum(~tthr)}")
+            print(f"number of pixels ruled out by d1 noise  = {np.sum(~k1thr)}")
+            print(f"number of pixels ruled out by d2 noise  = {np.sum(~k2thr)}")
+            print(f"number of bad pixels before threshold   = " f"{np.sum(~ok_wothr)}")
+        mask = mask | ~ok_pixel
+        T_e[mask] = 0.0
+        EM[mask] = 0.0
+        T_error[mask] = 0.0
+        EMerror[mask] = 0.0
+
+        if verbose:
+            print("from xrt_teem:")
+            Tmodel = tresp1.CHIANTI_temperature.value
+            if Trange:
+                Tmodel = Tmodel[
+                    (np.log10(Tmodel) >= Trange[0]) & (np.log10(Tmodel) <= Trange[1])
+                ]
+            print(f"Examined T_e range: {Tmodel.min()} - {Tmodel.max()} K")
+            print(f"Applied thresholds: - T_e error < {Te_err_threshold*100.} %")
+            print(
+                f"                    - Photon noise < "
+                f"{photon_noise_threshold*100.} %"
+            )
+    else:
+        if verbose:
+            Tmodel = tresp1.CHIANTI_temperature.value
+            print("from xrt_teem:")
+            print(f"Examined T_e range: {Tmodel.min()} - {Tmodel.max()} K")
+            print("No thresholds applied")
+    Tmap, EMmap, Terrmap, EMerrmap = make_results_maps(
+        hdr1, hdr2, T_e, EM, T_error, EMerror, mask
+    )
+    return Tmap, EMmap, Terrmap, EMerrmap
+
+
+def rebin_image(data, binfac=1):
```

Comment:
Perhaps we could rename this to `_rebin_image` so that it's not (yet) part of our public API?  I'm also wondering if this could go in `xrtpy.util`, since it's a generally useful capability.  That doesn't have to be in this PR, though.